Source/Rhetos.Deployment/DataMigrationScriptsFromDisk.cs

```diff
@@ -17,36 +17,59 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+using Newtonsoft.Json;
+using Rhetos.Extensibility;
 using Rhetos.Utilities;
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Text;
 using System.Text.RegularExpressions;
-using System.Threading.Tasks;
 
 namespace Rhetos.Deployment
 {
-    public class DataMigrationScriptsFromDisk : IDataMigrationScriptsProvider
+    public class DataMigrationScriptsFromDisk : IDataMigrationScriptsProvider, IGenerator
     {
+        const string DataMigrationScriptsFileName = "DataMigrationScripts.json";
         const string DataMigrationSubfolder = "DataMigration";
         const string DataMigrationSubfolderPrefix = DataMigrationSubfolder + @"\";
 
         protected readonly IInstalledPackages _installedPackages;
         private readonly FilesUtility _filesUtility;
+        private readonly RhetosAppEnvironment _rhetosAppEnvironment;
+        private List<DataMigrationScript> _scripts;
 
-        public DataMigrationScriptsFromDisk(IInstalledPackages installedPackages, FilesUtility filesUtility)
+        public IEnumerable<string> Dependencies => new List<string>();
+
+        public DataMigrationScriptsFromDisk(IInstalledPackages installedPackages,
+            FilesUtility filesUtility,
+            RhetosAppEnvironment rhetosAppEnvironment)
         {
             _installedPackages = installedPackages;
             _filesUtility = filesUtility;
+            _rhetosAppEnvironment = rhetosAppEnvironment;
         }
 
         /// <summary>
         /// The scripts are sorted by the intended execution order,
         /// respecting package dependencies and natural sort order by path.
         /// </summary>
         public List<DataMigrationScript> Load()
+        {
+            if (_scripts == null)
+            {
+                var dataMigrationScriptsFilePath = Path.Combine(_rhetosAppEnvironment.GeneratedFolder, DataMigrationScriptsFileName);

```

Comment:
Please add performance logger for this code block. "DataMigrationScriptsFromDisk: Loaded {_scripts.Count} scripts from generated file.", or something similar.